zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterProcess.java

```diff
@@ -51,13 +54,27 @@ public void setRemoteInterpreterEventPoller(RemoteInterpreterEventPoller eventPo
     this.remoteInterpreterEventPoller = eventPoller;
   }
 
+  public void shutdown()
+  {
+
+    // Close client socket connection
+    if (clientFactory != null)
+    {
+      for (TSocket eachTransfer: clientFactory.clientSocketMap.values())
+      {
+        eachTransfer.close();
+      }
+    }
+  }
+
   public int getConnectTimeout() {
     return connectTimeout;
   }
 
   public synchronized Client getClient() throws Exception {
     if (clientPool == null || clientPool.isClosed()) {
-      clientPool = new GenericObjectPool<>(new ClientFactory(getHost(), getPort()));
+      clientFactory = new ClientFactory(getHost(), getPort());
+      clientPool = new GenericObjectPool<>(clientFactory);
```

Comment:
This change is not necessary ?

Reply:
the Factory should be saved when create, use to call the close method

Reply:
I see.